src/Common/tests/System/Runtime/Serialization/Formatters/BinaryFormatterHelpers.cs

```diff
@@ -71,5 +72,46 @@ public static void AssertRoundtrips<T>(T expected, params Func<T, object>[] addi
                 }
             }
         }
+
+        public static void AssertExceptionDeserializationFails<T>() where T : Exception
+        {
+            // .NET Core and .NET Native throw PlatformNotSupportedExceptions when deserializing many exceptions.
+            // The .NET Framework supports full deserialization.
+            if (PlatformDetection.IsFullFramework)
+            {
+                return;
+            }
+
+            // Construct a valid serialization payload. This is necessary as most constructors call
+            // the base constructor before throwing a PlatformNotSupportedException, and the base
+            // constructor validates the SerializationInfo passed.
+            var info = new SerializationInfo(typeof(T), new FormatterConverter());
+            info.AddValue("ClassName", "ClassName");
+            info.AddValue("Message", "Message");
+            info.AddValue("InnerException", null);
+            info.AddValue("HelpURL", null);
+            info.AddValue("StackTraceString", null);
+            info.AddValue("RemoteStackTraceString", null);
+            info.AddValue("RemoteStackIndex", 5);
+            info.AddValue("HResult", 5);
+            info.AddValue("Source", null);
+            info.AddValue("ExceptionMethod", null);
+
+            // Serialization constructors are of the form .ctor(SerializationInfo, StreamingContext).
+            ConstructorInfo constructor = null;
+            foreach (ConstructorInfo c in typeof(T).GetTypeInfo().DeclaredConstructors)
```

Comment:
If any of these constructors are private on non-serializable types, .NET Native will prevent reflecting over them. I'm not sure which types you have in mind for this, but you may need to tolerate not finding a constructor in this loop.